Framework/jmaqs-selenium/src/main/java/com/magenic/jmaqs/selenium/WebDriverFactory.java

```diff
@@ -0,0 +1,554 @@
+    } catch (NumberFormatException e) {
+      throw new NumberFormatException("Length and Width must be a string that is an integer value: 400x400");
+    }
+  }
+
+  /**
+   * Get the web driver location.
+   *
+   * @param driverFile The web driver file, including extension
+   * @return The path to the web driver
+   */
+  public static String getDriverLocation(String driverFile) {
+    return getDriverLocation(driverFile, "", true);
+  }
+
+  /**
+   * Get the web driver location.
+   *
+   * @param driverFile      The web driver file, including extension
+   * @param defaultHintPath The default location for the specific driver
+   * @return The path to the web driver
+   */
+  public static String getDriverLocation(String driverFile, String defaultHintPath) {
+    return getDriverLocation(driverFile, defaultHintPath, true);
+  }
+
+  /**
+   * Get the web driver location.
+   *
+   * @param driverFile      The web driver file, including extension
+   * @param defaultHintPath The default location for the specific driver
+   * @param mustExist       Do we need to know where this drive is located,
+   *                        if this is true and the file is not found an error will be thrown
+   * @return The path to the web driver
+   */
+  public static String getDriverLocation(String driverFile, String defaultHintPath, boolean mustExist) {
+    // Get the hint path from the config
+    String hintPath = SeleniumConfig.getDriverHintPath();
+
+    // Try the hintpath first
+    if (!defaultHintPath.isEmpty() && Paths.get(hintPath, driverFile).toFile().exists()) {
+      return hintPath;
+    }
+
+    // Try the default hint path next
+    if (!defaultHintPath.isEmpty() && Paths.get(defaultHintPath, driverFile).toFile().exists()) {
+      return Paths.get(defaultHintPath).toString();
+    }
+
+    // Try the test location
+    Path path = Paths.get(new File("").getAbsolutePath());
+    String testLocation = path.getParent().toString();
+    if (Paths.get(testLocation, driverFile).toFile().exists()) {
+      return testLocation;
+    }
+
+    // Try resources
+    ClassLoader classLoader = WebDriverFactory.class.getClassLoader();
+    URL url = classLoader.getResource(driverFile);
+    if (url != null) {
+      File file = new File(url.getPath());
+      return file.getParent();
+    }
+
+    // We didn't find the web driver so throw an error if we need to know where it is
+    if (mustExist) {
+      throw new RuntimeException(StringProcessor.safeFormatter("Unable to find driver for '%s'", driverFile));
+    }
+
+    return "";
+  }
+
+  /**
+   * Get the programs file folder which contains given file (for finding the Edge driver).
+   *
+   * @param folderName The programs file sub folder
+   * @param file       The file we are looking for
+   * @return The parent folder of the given file or the empty string if the file is not found
+   */
+  static String getProgramFilesFolder(String folderName, String file) {
```

Comment:
This is Edge only so can we get a more descriptive name? Otherwise the Windows specific logic might throw us for a loop when we move to more cross platform support.

Reply:
I've renamed this to getWindowsEdgeDriverLocation